tests/test_least_squares.py

```diff
@@ -0,0 +1,54 @@
+from lmfit import Parameters, minimize, fit_report, Minimizer
+from lmfit_testutils import assert_paramval, assert_paramattr
+
+from numpy import linspace, zeros, sin, exp, random, pi, sign
+
```

Comment:
Will do a test for Nr. 1. Nr. 3 is not yet done, so nothing to test yet. 
Can you specify what you mean by output? Because testing if the parameters
converge to same result is done indirectly: we compare with the true result.